src/modules/networks/Networks.ts

```diff
@@ -25,152 +25,75 @@ function toCollection<T, K>(
+    positions: 'supports_positions',
   } as const;
 
   constructor({
     networks,
-    ethereumChainSources,
+    ethereumChainConfigs,
   }: {
     networks: NetworkConfig[];
-    ethereumChainSources?: EthereumChainSources;
+    ethereumChainConfigs: EthereumChainConfig[];
   }) {
-    this.networks = networks
-      .filter((network) => Boolean(network.evm_id))
-      .sort((a, b) => localeCompareWithPriority(a.name, b.name, 'Ethereum'));
-    this.ethereumChainSources = ethereumChainSources;
-    const {
-      collection,
-      originalCollection,
-      collectionByEvmId,
-      sourcesNormalized,
-    } = this.prepare();
-    this.collection = collection;
-    this.originalCollection = originalCollection;
-    this.collectionByEvmId = collectionByEvmId;
-    this.sourcesNormalized = sourcesNormalized;
-    this.supportedByBackend = new Set(networks.map((n) => n.chain));
-  }
-
-  private prepare() {
-    const originalCollection = toCollection(
+    this.ethereumChainConfigs = ethereumChainConfigs;
+    this.networks = insertChainConfigs(networks, ethereumChainConfigs);
+    this.collection = toCollection(
       this.networks,
-      (network) => network.chain,
+      (network) => network.id,
       (x) => x
     );
-    const collectionByEvmId = toCollection(
+    this.collectionByEvmId = toCollection(
       this.networks,
-      (x) => x.external_id,
+      (x) => getChainId(x) || '',
       (x) => x
-    );
-    const sourcesNormalized = normalizeSources(this.ethereumChainSources);
-    const collection = { ...originalCollection };
-    for (const value of Object.values(sourcesNormalized)) {
-      if (!value) {
-        continue;
-      }
-      Object.assign(collection, value.collection);
-      Object.assign(
-        collectionByEvmId,
-        toCollection(
-          value.items,
-          (item) => item.external_id,
-          (item) => item
-        )
-      );
-    }
-    return {
-      collection,
-      originalCollection,
-      collectionByEvmId,
-      sourcesNormalized,
-    };
-  }
-
-  updateEthereumChainSources(ethereumChainSources: EthereumChainSources) {
-    this.ethereumChainSources = ethereumChainSources;
-    const {
-      collection,
-      collectionByEvmId,
-      originalCollection,
-      sourcesNormalized,
-    } = this.prepare();
-    this.collection = collection;
-    this.originalCollection = originalCollection;
-    this.collectionByEvmId = collectionByEvmId;
-    this.sourcesNormalized = sourcesNormalized;
+    ) as { [key: number]: NetworkConfig | undefined };
   }
 
   static getName(network: NetworkConfig) {
-    return network.name || capitalize(network.chain);
+    return network.name || capitalize(network.id);
   }
 
   private toId(chain: Chain) {
     const item = this.collection[chain.toString()];
     if (!item) {
       throw new Error(`Chain not found: ${chain}`);
     }
-    return item.external_id as ChainId;
+    return getChainId(item);
+  }
+
+  isSavedChain(chain: Chain) {
```

Comment:
Let's discuss how to name this one better

Reply:
Sure!

Reply:
Updated it, but still not so sure